eo-maven-plugin/src/main/java/org/eolang/maven/name/OnVersioned.java

```diff
@@ -42,15 +43,24 @@ public final class OnVersioned implements ObjectName {
     /**
      * Default hash.
      */
-    private final CommitHash hsh;
+    private final Unchecked<CommitHash> hsh;
```

Comment:
@volodya-lombrozo [here](https://github.com/objectionary/eo/pull/2432/files/b6486169dfaecf9a6ae3cd6e7acc8bde46bed540#r1309997317). Scalar is needed because `new OnDefault(tojo).hash()` may fail with `IndexOutOfBounds` exception if tojo does not contain delimiter `|`

Reply:
@maxonfjvipon I see, thank you. Maybe it's better to let it fail as fast as possible? Otherwise it looks like you are trying to swallow the exception or use exceptions for flow control (both of the approaches don't lead to a good design).

Reply:
@volodya-lombrozo It should not fail at this step at all. Pay a bit more attention to the code:
```java
new OnSwap(
    tojo.contains(OnReplaced.DELIMITER),
    new OnVersioned(
        replaced,
        new OnDefault(tojo)::hash
    ),
    replaced
)
```
If condition `tojo.contains(OnReplaced.DELIMITER)` is FALSE, the first object will be never evaluated, the second one will be - `replaced`. But we need to construct it here and it should now fail on the construction step

Reply:
@volodya-lombrozo maybe we will add one more object name for such cases, but there's only one for now, so there's no need